lib/HLSL/DxilGenerationPass.cpp

```diff
@@ -1364,9 +1364,8 @@ static void PropagatePreciseAttributeOnOperand(Value *V, DxilTypeSystem &typeSys
     return;
   // TODO: skip precise on integer type, sample instruction...
 
-  // Set precise fast math on those instructions that support it.
-  if (DxilModule::PreservesFastMathFlags(I))
```

Comment:
We added this because we can’t round trip the assembly if we set fast math everywhere. I think we need it

Reply:
Here what we're doing is remove the fast math. Not add it.

Reply:
But if the instruction does not preserve precise then why are we setting it here? Does that function have a bug?

Reply:
But if the instruction does not preserve precise then why are we setting it here? Does that function have a bug?